### ISSUE TITLE
disable eslint in vue template

### ISSUE THREAD

**Issue by siavashkavousi** [vuejs/eslint-plugin-vue#260]
Hi,

Is there anyway to disable lint in `template` like in `js` (e.g. `eslint-disable-line`)?

**Comment by mysticatea** [comment]
Thank you for this issue.

There is no way, currently.
We should investigate the way to implement directive comments such as `eslint-disable`.

**Comment by MaxMilton** [comment]
It's particularly frustrating with the `max-len` eslint rule. I often have paragraphs or long elements and so I've had to disable the rule. A shame because it's useful for JS linting.

**Comment by asifm** [comment]
Any way to disable eslint in pug templates? The solution given apparently doesn't work with pug.

**Comment by 1c7** [comment]
2018 June 13, 
this annoying me, so I just search and remove `js-lint`

**Comment by Julia-Lavrova** [comment]
You need to add ':key="t"', it needs for circle's item.

**Comment by 1c7** [comment]
@Julia-Lavrova Thanks!

**Comment by maximusblade** [comment]
@Julia-Lavrova  thanks.

**Comment by willin** [comment]
@mysticatea  how about now???


```html
    <!-- eslint disable next-line vue/no-v-html -->
    <p class="post-excerpt" v-html="post.excerpt || post.frontmatter.description || '' " />
```

**Comment by benjaminpwarren** [comment]
The above `vue/no-v-html` example works for me, but `<!-- eslint-disable vue/valid-v-on -->` doesn't, the error still shows. Using 5.0.0-beta.3.

**Comment by KITSMoeNashashibi** [comment]
try `<!-- eslint-disable -->` 
this will disable linting for everything below it
you can use `<!-- eslint-enable -->` to re-enable it back

**Comment by aniude** [comment]
@willin  It seems that it can't resolved this situation
 ```
<p class="post-excerpt"
      v-html="post.excerpt || post.frontmatter.description || '' " />
```

**Comment by EdwinBetanc0urt** [comment]
your_project/config/index.js

change the value of **useEslint** from true to false, from line 26.

Note: I use the vue webpack template, the version of the template is 1.3.1

**Comment by blowsie** [comment]
@aniude you can use it like so;

```
        <!-- eslint-disable vue/no-v-html -->
        <div
          v-if="!manageOptions"
          class="content"
          v-html="$t('gdpr.marketingConsent.body')"
        />
        <!--eslint-enable-->
```

**Comment by blowsie** [comment]
@gluons you need to use it inside the `<template>` tag, this works on webstorm at least

**Comment by gluons** [comment]
@blowsie It still won't work. Not only VSCode. Vue CLI throw the errors.

![default](https://user-images.githubusercontent.com/4688092/50588892-bed55700-0eb6-11e9-96ec-553fa457b7d2.png)

![default](https://user-images.githubusercontent.com/4688092/50588898-c98fec00-0eb6-11e9-8f03-628e01b9e039.png)

**Comment by michalsnik** [comment]
We don't support pug templates @gluons Depending on version of `vue-eslint-parser` you have installed templates other than `html` should be ignored. Can you provide informations about versions `eslint-plugin-vue` and `vue-eslint-parsers` you use?

**Comment by gluons** [comment]
Just remove `yarn.lock` and `node_modules` and reinstall modules. Here is it.

**`eslint-plugin-vue`:**
```bash
> yarn list --pattern eslint-plugin-vue
yarn list v1.12.3
├─ @vue/cli-plugin-eslint@3.2.2
│  └─ eslint-plugin-vue@4.7.1
└─ eslint-plugin-vue@5.1.0
```

**`vue-eslint-parser`:**
```bash
> yarn list --pattern vue-eslint-parser
yarn list v1.12.3
├─ @vue/cli-plugin-eslint@3.2.2
│  └─ vue-eslint-parser@2.0.3
└─ vue-eslint-parser@4.0.3
```

---

Errors still exist.

![default](https://user-images.githubusercontent.com/4688092/50597515-3799da80-0eda-11e9-9bee-c108a960c63f.png)

![default](https://user-images.githubusercontent.com/4688092/50597548-55673f80-0eda-11e9-92c4-586ed10288fd.png)

**Comment by PayteR** [comment]
Is there a way to ignore someone specific rule globally in the .eslintrc file? I tried this, but it doesn't work

```
module.exports = {
  root: true,
  env: {
    browser: true,
    node: true
  },
  parserOptions: {
    parser: 'babel-eslint'
  },
  extends: [
    '@nuxtjs',
    'plugin:nuxt/recommended',
    'plugin:prettier/recommended',
    'prettier',
    'prettier/vue'
  ],
  plugins: [
    'prettier'
  ],
  // add your custom rules here
  rules: {
    "vue/no-v-html": "off"
  }
};
```
I want this, because i want to use v-html everywhere and don't need to set ignore on every occurence.

**Comment by nashashibi** [comment]
I'd try `'vue/no-v-html': true | false` first.. (lucky guess)

you can check this link as well, hopefully you'll find some useful answers for your issue:
[https://github.com/vuejs-templates/webpack/issues/73](url)

**Comment by bigsee** [comment]
@PayteR I think this rule expects an int rather than a string. I have it working like this:

```
module.exports = {
  root: true,
  env: {
    browser: true,
    node: true
  },
  parserOptions: {
    parser: 'babel-eslint'
  },
  extends: [
    '@nuxtjs',
    'plugin:nuxt/recommended'
  ],
  // add your custom rules here
  rules: {
    'nuxt/no-cjs-in-config': 'off',
    'vue/no-v-html': 0,
    'no-console': 0
  }
}
```

**Comment by iigmir** [comment]
The error exists in VSCode even I try to ignore it:
```
<div>
    <!--eslint-disable vue/valid-v-for-->
    <component v-for="(item, index) in lists" v-bind="component_binding(item, index)" v-on:event_com="event_binding"></component>
    <!--eslint-enable-->
</div>
```
Software version:
1. VSCode v.1.42
2. eslint-plugin-html v.6.0.0
3. eslint-plugin-vue v.5.2.3

**Comment by Guley** [comment]
> 
> 
> try `<!-- eslint-disable -->`
> this will disable linting for everything below it
> you can use `<!-- eslint-enable -->` to re-enable it back

Add in App.vue
<script> /* eslint-disable */ </script>

**Comment by sirlancelot** [comment]
I'm using eslint-plugin-vue@6.2.2 and I'm still experiencing this bug:
![image](https://user-images.githubusercontent.com/71916/88866555-e4743080-d1bf-11ea-810b-8fc1e193a05f.png)
Interestingly, without the flag on line 75, VSCode actually emits TWO warnings, one from ESLint, and the other somehow from this plugin...
![image](https://user-images.githubusercontent.com/71916/88866635-1d140a00-d1c0-11ea-8b5b-ec4987078b83.png)
I need to ignore it for this line only (not the whole file) as my `cellData` variable contains a `key`.

**Comment by KITSMoeNashashibi** [comment]
why not just pass `:key="cellData.key"`?

**Comment by sirlancelot** [comment]
> why not just pass `:key="cellData.key"`?

That is what I've done as a workaround, but it doesn't make this not a bug.

**Comment by rajeshwarpatlolla** [comment]
It's not working for me. I tried to disable in template and also tried with eslint config in package.json. Nothing is working for me.

<img width="607" alt="Screenshot 2020-08-05 at 1 20 27 AM" src="https://user-images.githubusercontent.com/9896335/89338573-61b90d00-d6ba-11ea-923a-c0b3c22dd0be.png">

<img width="578" alt="Screenshot 2020-08-05 at 1 19 59 AM" src="https://user-images.githubusercontent.com/9896335/89338602-6bdb0b80-d6ba-11ea-95ed-8c6629fd0d8b.png">

<img width="326" alt="Screenshot 2020-08-05 at 1 25 06 AM" src="https://user-images.githubusercontent.com/9896335/89338658-82816280-d6ba-11ea-8735-ccd5d6ca9187.png">

The error is irritating. Please let me know if anyone has a solution for this.

**Comment by iigmir** [comment]
> why not just pass `:key="cellData.key"`?

Because you can't do this:

```html
<div>
    <component :="{ key: cellData.key, class: cellData.class }"></component>
</div>
```

In this situation, you must do something like this:

```html
<div>
    <component v-bind="{ key: cellData.key, class: cellData.class }"></component>
</div>
```

**Comment by chasedougherty4628** [comment]
> It's not working for me. I tried to disable in template and also tried with eslint config in package.json. Nothing is working for me.
> 
> <img alt="Screenshot 2020-08-05 at 1 20 27 AM" width="607" src="https://user-images.githubusercontent.com/9896335/89338573-61b90d00-d6ba-11ea-923a-c0b3c22dd0be.png">
> 
> <img alt="Screenshot 2020-08-05 at 1 19 59 AM" width="578" src="https://user-images.githubusercontent.com/9896335/89338602-6bdb0b80-d6ba-11ea-95ed-8c6629fd0d8b.png">
> 
> <img alt="Screenshot 2020-08-05 at 1 25 06 AM" width="326" src="https://user-images.githubusercontent.com/9896335/89338658-82816280-d6ba-11ea-8735-ccd5d6ca9187.png">
> 
> The error is irritating. Please let me know if anyone has a solution for this.

@rajeshwarpatlolla let me know if you find a solution. I have the exact same problem

**Comment by ikubij** [comment]
I have the same problem, when using vee validate...Its v-slots

**Comment by nuckles30** [comment]
Try this
<div v-for="(thread, index) in threads" :key="index">

**Comment by nuckles30** [comment]
try this
`<div v-for="(thread, index) in threads" :key="index">`

**Comment by ttntm** [comment]
I've got `"vue/custom-event-name-casing": "off"` in the eslint rules section of my `package.json` but it doesn't seem to have any effect. 

I'd really like to get rid of it, as it's quite annoying to have all my `$emit('property:update', val)` events marked as an error in VS Code, even though as per vue docs this is actually valid (+ working perfectly).

**Comment by Hijiri-K** [comment]
@chasedougherty4628 

I had the same problem.
But I could solve with adding a line bellow into `.vscode/setting.json`.
```
"vetur.validation.template": false
```
reference: https://gitmemory.com/issue/vuejs/eslint-plugin-vue/1269/668921391
 
If you are using `eslint` with `vscode` and `vetur`, it may help you.

**Comment by yubaoquan** [comment]
![image](https://user-images.githubusercontent.com/2340296/97960708-eec04f00-1dec-11eb-869d-26d6a3a858c3.png)
Same issue

**Comment by MatthewZito** [comment]
All of the suggestions thus far weren't working for me either until I tried enclosing an actual C-style comment block _within_ the html comment. This works for me:

```
<template lang="pug">
<!-- /* eslint-disable */ -->
...
```

Hopefully this works for others. Let us know!

**Comment by woai3c** [comment]
`/* eslint-disable max-len */` -> `<!-- eslint-disable max-len -->` , this is work for me.

![image](https://user-images.githubusercontent.com/22117876/134026623-bb16c20c-379b-4e7b-9c03-9833c5cedd7f.png)

**Comment by jez9999** [comment]
The only problem is, those comments will show up in the DOM:

> 

We need a way to hide them from the DOM!

**Comment by jez9999** [comment]
I guess that works, but only if you want to disable the rule for the entire template.

**Comment by Vlasterx** [comment]
There is a problem with `vue/valid-v-on` rule when adding custom modifiers.

Let's say that I have created these modifiers for `keydown`:
```js
export const Keycodes = {
	multiply: 106,
	minus: 109,
	plus: 107,
	divide: 111,
}
``` 

If I initialize these custom keycodes with 
```js
Vue.config.keyCodes = Keycodes
```

and add them to `eslint.js` config like this:

```js
const { Keycodes } = require('./Keycodes')
const customKeycodes = Object.keys(Keycodes)

module.exports = {
  //rest of the eslint settings
  'rules': {
    // Rest of eslint rules
    'vue/valid-v-on': ['warn', {'modifiers': customKeycodes }],
  }
}
```

It doesn't work in VSCode, but the webpack is compiling without errors.

Problem is that I cannot report this error, since it is in a private repository. :-/
